app/components/primer/alpha/overlay/header.rb

```diff
@@ -13,6 +13,18 @@ class Header < Primer::Component
         }.freeze
         SIZE_OPTIONS = SIZE_MAPPINGS.keys
 
+        # Optional filter slot for adding a filter input to the header.
+        #
+        # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+        renders_one :filter, lambda { |**system_arguments|
```

Comment:
Since we're naming it `filter`, what would you think of the slot rendering the `TextField` instead of asking callers to fill it in?

Reply:
It's a tough call, cause the current usage I need the content because I'm running into different implementations of "filter" I could see next quarter as we consolidate them maybe we do restrict it